src/ui/DynamicFacet/DynamicRangeFacet.ts

```diff
@@ -0,0 +1,128 @@
+
+  static options: IDynamicRangeFacetOptions = {
+    /**
+     * The separator between the minimum and the maximum.
+     *
+     * **Default:** The localized string for `to`.
+     */
+    valueSeparator: ComponentOptions.buildLocalizedStringOption({ defaultValue: l('To'), section: 'CommonOptions' }),
+    /**
+     * The format used for the minimum and maximum value.
+     * Possible values are: `number`, `date` and `currency`
+     *
+     * **Default:** `number`
+     */
+    valueFormat: ComponentOptions.buildStringOption({ defaultValue: DynamicRangeFacetValueFormat.number, section: 'CommonOptions' }),
+    /**
+     * The list of [range values]{@link IRangeValue} to request (see [Requesting Specific FacetRange Values](https://docs.coveo.com/en/2790/)).
+     *
+     * By default, the index automatically generates range values.
+     *
+     * **Note:**
+     * > The index cannot automatically generate range values for a `DynamicRangeFacet` whose [`field`]{@link DynamicRangeFacet.options.field}
+     * > option value references a dynamic field generated by a [query function](https://docs.coveo.com/en/232/).
+     * > In such a case, you _must_ use the `ranges` option.
+     */
+    ranges: ComponentOptions.buildJsonOption<IRangeValue[]>()
+  };
+
+  /**
+   * Creates a new `DynamicRangeFacet` instance.
+   *
+   * @param element The element from which to instantiate the component.
+   * @param options The component options.
+   * @param bindings The component bindings. Automatically resolved by default.
+   */
+  constructor(public element: HTMLElement, public options: IDynamicRangeFacetOptions, bindings?: IComponentBindings) {
+    super(element, ComponentOptions.initComponentOptions(element, DynamicRangeFacet, options), bindings, DynamicRangeFacet.ID);
+
+    this.disableUnavailableOptions();
+  }
+
+  protected initValues() {
+    this.values = new DynamicFacetValues(this, DynamicRangeFacetValueCreator);
+
+    if (this.options.ranges) {
+      this.values.createFromRanges(this.options.ranges);
+    }
+  }
+
+  protected initDynamicFacetQueryController() {
+    this.dynamicFacetQueryController = new DynamicRangeFacetQueryController(this);
+  }
+
+  private disableUnavailableOptions() {
+    this.options.enableFacetSearch = false;
+    this.options.useLeadingWildcardInFacetSearch = false;
+    this.options.enableMoreLess = false;
+    this.options.valueCaption = {};
+  }
+
+  public get facetType(): FacetType {
+    if (this.options.valueFormat === DynamicRangeFacetValueFormat.date) {
+      return FacetType.dateRange;
+    }
+
+    return FacetType.numericalRange;
+  }
+
+  public showMoreValues() {
+    this.logger.warn('The "showMoreValues" method is not available on the "DynamicRangeFacet" component');
+  }
+
+  public showLessValues() {
+    this.logger.warn('The "showLessValues" method is not available on the "DynamicRangeFacet" component');
+  }
+
+  public async triggerNewIsolatedQuery() {
```

Comment:
When I see methods that are not implemented, it makes me question the inheritance tree, and whether we are forcing a model that does not apply. It's good to check whether the quantity of code we reuse is worth the methods we stub and override using `protected` key word.

If it's not a good return, it would be better to avoid the inheritence and either:

1. Repeat what (little) needs to be repeated,
2. Compose a Dynamic Facet and call what you need, or
3. Extract logic into a shared utility class and reuse the functions in both classes.

Reply:
You're right that inheritance is not the best solution in some cases. I really tested the idea of doing composition however the amount the code that would need to be rewritten or extracted was too much and would have complexified the code too much.

Instead disabling some options and overriding methods seemed the best options, the cost there is low and everything is still rather clean.
Note that the 3 methods are 
1. showMoreValues
2. showLessValues
3. triggerNewIsolatedQuery (basically search)
In the future we could decide to extend ranges (e.g. enabling the show more values) thats a use case I saw in some ecommerce websites
For triggerNewIsolatedQuery, we could allow a user to customize his own range or there might be some cases for a search that a client could come up with.. So it's not really wasted